experimental/beacon_sim/correlated_beacons.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import numpy as np
+from typing import NamedTuple
+from scipy import optimize, special
+import math
+
+
+class BeaconClique(NamedTuple):
+    """
+    This represents a set of beacons that are correlated in appearance
+    """
+    # The marginal probability of a beacon
+    p_beacon: float
```

Comment:
will beacons in a clique ever differ in their marginals?

Also missed that p_no_beacon != 1-p_beacon until I was reading through the tests. Making it `p_no_beacons` may clarify this but doesn't seem critical

Reply:
Currently no, but it could be updated in the future.

Updated the comment and variable name. Thank you for the suggestion.